backend/Services/ReservationService.cs

```diff
@@ -0,0 +1,151 @@
+            PricePerNight = 8m,
+            RoomTemplateID = Guid.NewGuid(),
+        },
+        new Room()
+        {
+            RoomID = _reservations[1].RoomID,
+            Code = "A2",
+            FloorNumber = 124,
+            HotelID = Guid.NewGuid(),
+            PricePerNight = 9m,
+            RoomTemplateID = Guid.NewGuid(),
+        }
+    };
+    
+    private ReservationConverter _converter = new ReservationConverter();
+    public override async Task<ReservationDTO> GetReservationById(Guid reservationId)
+    {
+        await Task.Delay(10);
+        var reservation = _reservations.FirstOrDefault(x => x.ContactID == reservationId);
+        if (reservation == null)
+        {
+            throw new Exception("Reservation not found");
+        }
+        var contact = _contacts.FirstOrDefault(x => x.ContactID == reservation.ContactID);
+        var room = _rooms.FirstOrDefault(x => x.RoomID == reservation.RoomID);
+        return _converter.Convert(reservation, contact, room);
+    }
+
+    public override async Task<List<ReservationDTO>> GetReservations()
+    {
+        await Task.Delay(10);
+        List<ReservationDTO> result = _reservations.Select(r =>
+        {
+            var contact = _contacts.FirstOrDefault(x => x.ContactID == r.ContactID);
+            var room = _rooms.FirstOrDefault(x => x.RoomID == r.RoomID);
+            return _converter.Convert(r, contact, room);
+        }).ToList();
+        
+        return result;
+    }
+
+    public override async Task<ReservationDTO> CreateReservation(ReservationDTO reservationDto)
+    {
+        await Task.Delay(100);
+        if (reservationDto != null)
+        {
+            var newReservation = new Reservation
+            {
+                ContactID = Guid.NewGuid(),
+                Cancelled = reservationDto.Cancelled,
+                ReservationDate = reservationDto.ReservationDate,
+                RoomID = Guid.NewGuid(),
+                UseDate = reservationDto.UseDate,
+            };
+            _reservations.Add(newReservation);
+
+            var newContact = new Contact
+            {
+                ContactID = newReservation.ContactID,
+                Email = reservationDto.UserEmail,
+                PhoneNumber = reservationDto.UserPhoneNumber,
+            };
+            _contacts.Add(newContact);
+
+            var newRoom = new Room
+            {
+                RoomID = reservationDto.RoomID,
+                Code = reservationDto.RoomCode,
+                FloorNumber = reservationDto.RoomFloorNumber,
+                HotelID = Guid.NewGuid(),
+                PricePerNight = reservationDto.PricePerNight
+            };
+            _rooms.Add(newRoom);
+            if(_reservations.Contains(newReservation) && _contacts.Contains(newContact) && _rooms.Contains(newRoom))
+                return reservationDto;
+            else 
+                throw new Exception("Reservation not created");
+        }
```

Comment:
you must add an else case to the first if

Reply:
I add that line